src/main/java/com/synopsys/integration/jenkins/scan/global/Utility.java

```diff
@@ -58,4 +65,76 @@ public static void removeFile(String filePath, FilePath workspace, TaskListener
     public static boolean isStringNullOrBlank(String str) {
         return str == null || str.isBlank() || str.equals("null");
     }
+
+    public static HttpURLConnection getHttpURLConnection(URL url, EnvVars envVars, LoggerWrapper logger) {
+        try {
+            String proxy = getProxy(url, envVars);
+            if (proxy.equals(ApplicationConstants.NO_PROXY)) {
+                return (HttpURLConnection) url.openConnection(Proxy.NO_PROXY);
+            } else {
+                URL proxyURL = new URL(proxy);
+
+                HttpURLConnection connection = (HttpURLConnection) url.openConnection(
+                    new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyURL.getHost(), proxyURL.getPort())));
+                setDefaultProxyAuthenticator(proxyURL.getUserInfo());
+
+                return connection;
+            }
+        } catch (IOException e) {
+            logger.error("An exception occurred while getting HttpURLConnection: " + e.getMessage());
+        }
+
+        return null;
+    }
+
+    public static String getProxy(URL url, EnvVars envVars) {
+        String noProxy = getEnvOrSystemProxyDetails(ApplicationConstants.NO_PROXY, envVars);
+        if (!isStringNullOrBlank(noProxy)) {
+            if (noProxy.contains(url.toString())) {
+                return ApplicationConstants.NO_PROXY;
```

Comment:
Do we need to take care of wildcard as well. We should discuss this.

Reply:
As discussed, wildcard no_proxy support is not required for first release.